cctp_primitives/src/proving_system/init.rs

```diff
@@ -18,104 +13,204 @@ use std::sync::{RwLock, RwLockReadGuard};
 // additionally wrapped the committer key in a RwLock.
 
 lazy_static! {
-    pub static ref G1_COMMITTER_KEY: RwLock<Option<CommitterKeyG1>> = RwLock::new(None);
+    pub static ref G1_UNIVERSAL_PARAMS: RwLock<Option<UniversalParams<G1>>> = RwLock::new(None);
 }
 
 lazy_static! {
-    pub static ref G2_COMMITTER_KEY: RwLock<Option<CommitterKeyG2>> = RwLock::new(None);
+    pub static ref G2_UNIVERSAL_PARAMS: RwLock<Option<UniversalParams<G2>>> = RwLock::new(None);
 }
 
-/// Generate G1CommitterKey and store it in memory.
+/// Generate `G1_UNIVERSAL_PARAMETERS` and store it in memory.
+/// This function should be called exactly once during program execution and before any call to
+/// `get_g1_committer_key()`. Further calls leave `G1_UNIVERSAL_PARAMETERS` unchanged and return an
+/// error instead.
 /// The parameter `max_degree` is required in order to derive a unique hash for the key itself.
-pub fn load_g1_committer_key(
-    max_degree: usize,
-    supported_degree: usize,
-) -> Result<(), SerializationError> {
-    match load_generators::<G1>(max_degree, supported_degree) {
+pub fn load_g1_universal_params(max_degree: usize) -> Result<(), SerializationError> {
```

Comment:
Please let's keep the old name anyway to be sure not to break dependant libraries (formally, doesn't change anything anyway)